web/simul_write.go

```diff
@@ -0,0 +1,59 @@
+package web
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"sync"
+
+	"github.com/criteo/graphite-remote-adapter/client"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/common/model"
+)
+
+// SimulWriteRequest
+type SimulWriteRequest struct {
+	Samples []*model.Sample `json:"samples"`
+}
+
+// SimulWriteResponse
+type SimulWriteResponse struct {
+	Writers map[string]string `json:"writers"`
+}
+
+func (h *Handler) simulWrite(w http.ResponseWriter, r *http.Request) {
+	h.lock.RLock()
```

Comment:
Looks like some of `Write` is duplicate, maybe this should be the same function with a boolean flag ?

Reply:
The 2 function don't take the same kind of request as argument (simulWrite take a json to make it a more usable api, and Write a compressed protobuf to be compliant with prometheus).
Therefore the remaining common code is a for loop over writers client.

Reply:
as discussed we could use a single function that look at content type to parse the request